helper.go

```diff
@@ -12,3 +12,18 @@ type noopHelperTT struct {
 func (*noopHelperTT) Helper() {
 	// Intentional NOOP
 }
+
+// deepEqualityPrinter is a utility Printer that lets the jsonassert package
+// verify equality internally without affecting the external facing output,
+// e.g. to verify where one potential candidate is matching or not.
+type deepEqualityPrinter struct{ count int }
+
+func (p *deepEqualityPrinter) Errorf(msg string, args ...interface{}) { p.count++ }
+func (p *deepEqualityPrinter) Helper()                                { /* Intentional NOOP */ }
+
+func (a *Asserter) deepEqual(act, exp interface{}) bool {
```

Comment:
The performance here is probably *significantly* worse, depending on just how much nesting is going on. Performance is explicitly not a goal of this package, but there may be some wins around doing a naive `deepEqual` like before and only doing this more expensive check in case of failures only. Would need to benchmark etc., so will leave this as is for now. Also not committing to ever fixing the perf in the future.

Reply:
Actually the above would be a tradeoff b/w breadth/depth. Indeed, benchmarking would be required before implementing anything like that.